File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        d = dummy = ListNode()
        prev, cur = None, head
        while cur:
            if cur.val != 0:
                if prev != None:
                    prev.val += cur.val
                else:
                    prev = cur
                    d.next = prev
                    d = d.next
            else:
                prev = None
            cur = cur.next
        d.next = None
        return dummy.next

File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var mergeNodes = function(head) {
    const dummy = new ListNode()
    let temp = dummy
    let next = null
    for (let cur=head; cur; cur=cur.next) {
        if (cur.val !== 0) {
            if (next !== null) {
                next.val += cur.val
            } else {
                next = cur
                temp.next = next
                temp = temp.next
            }
        } else {
            next = null
        }
    }
    temp.next = null
    return dummy.next
};